shapes/shape.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <QPoint>
+#include <QString>
+#include <string>
+
+namespace jbrush
+{
+class Shape
+{
+public:
+    /* Constructors */
+    Shape(QPoint);
+
+    /* Destructor */
+    virtual ~Shape();
+
+    /* Position */
+    QPoint getPosition() const;
+    void   setPosition(QPoint);
+
+    /* Render */
+    virtual void draw() const = 0;
```

Comment:
This definitely needs to have a QWidget parameter so the Shape knows what surface to draw to